data/service/api/v1/summary.go

```diff
@@ -0,0 +1,113 @@
+package v1
+
+import (
+	"net/http"
+    "strconv"
+
+	//"github.com/tidepool-org/platform/data"
+	dataService "github.com/tidepool-org/platform/data/service"
+	//"github.com/tidepool-org/platform/page"
+	//"github.com/tidepool-org/platform/permission"
+	"github.com/tidepool-org/platform/request"
+)
+
+func SummaryRoutes() []dataService.Route {
+	return []dataService.Route{
+		dataService.MakeRoute("GET", "/v1/summary/:userId", Authenticate(GetSummary)),
+		dataService.MakeRoute("POST", "/v1/summary/:userId", Authenticate(UpdateSummary)),
+		dataService.MakeRoute("GET", "/v1/agedsummaries/:minutes", Authenticate(GetAgedSummaries)),
+	}
+}
+
+func GetSummary(dataServiceContext dataService.Context) {
+	res := dataServiceContext.Response()
+	req := dataServiceContext.Request()
+	dataClient := dataServiceContext.DataClient()
+
+	//details := request.DetailsFromContext(req.Context())
+	/*if details == nil {
+		request.MustNewResponder(res, req).Error(http.StatusUnauthorized, request.ErrorUnauthenticated())
+		return
+	}*/
+
+	responder := request.MustNewResponder(res, req)
+
+	id := req.PathParam("userId")
+	if id == "" {
```

Comment:
Curiosity: I've seen this pattern all over the existing service code, but never asked if it is actually needed.
The route declaration already "knows" that this path should include `:userId`, and therefore missing it _ought_ to be grounds for a 404.

Reply:
This is technically true in this case, as %20 is never unencoded into an empty arg while still matching the url.
block removed.